frontend/packages/dev-console/src/components/import/__tests__/DeployImage.spec.tsx

```diff
@@ -0,0 +1,90 @@
+import * as React from 'react';
+import { mount, ReactWrapper } from 'enzyme';
+import { Provider } from 'react-redux';
+import { Radio } from '@patternfly/react-core';
+import store from '@console/internal/redux';
+import { PageHeading, ButtonBar } from '@console/internal/components/utils/';
+import NamespacedPage from '../../NamespacedPage';
+import DeployImagePage from '../DeployImagePage';
+import DeployImage from '../DeployImage';
+import ImageSearchSection from '../image-search/ImageSearchSection';
+import AppSection from '../app/AppSection';
+import AdvancedSection from '../advanced/AdvancedSection';
+import ResourceSection from '../section/ResourceSection';
+
+describe('DeployImage Page Test', () => {
+  type DeployImagePageProps = React.ComponentProps<typeof DeployImagePage>;
+  const deployImagePageProps: DeployImagePageProps = {
+    history: null,
+    location: {
+      pathname: 'deploy-image/ns/openshift?preselected-ns=openshift',
+      search: 'deploy-image/ns/openshift?preselected-ns=openshift',
+      state: null,
+      hash: null,
+    },
+    match: {
+      isExact: true,
+      path: 'deploy-image/ns/openshift?preselected-ns=openshift',
+      url: 'deploy-image/ns/openshift?preselected-ns=openshift',
+      params: {
+        ns: 'my-project',
+      },
+    },
+  };
+  const deployImagePageWrapper: ReactWrapper = mount(
+    <DeployImagePage {...deployImagePageProps} />,
+    {
+      wrappingComponent: ({ children }) => <Provider store={store}>{children}</Provider>,
+    },
+  );
```

Comment:
should put this setup in `beforeAll()` block and declare the variables in describe block

Reply:
The props initialization is not working when put in beforeEach/beforeAll? what is wrong with the existing scoping?

Reply:
All the setup for unit-tests should be done in either `beforeAll` or `beforeEach` depending on the usage. it works when put in `beforeAll` block.

Reply:
I've updated...